File: app/src/main/java/com/smartsoft/pereverzev/dagger2oldtutorial/screens/HomeActivityComponent.java
package com.smartsoft.pereverzev.dagger2oldtutorial.screens;


import com.smartsoft.pereverzev.dagger2oldtutorial.GithubApplicationComponent;

import dagger.Component;

@HomeActivityScope
@Component(modules = HomeActivityModule.class, dependencies = GithubApplicationComponent.class)
public interface HomeActivityComponent {

    void injectHomeActivity(HomeActivity homeActivity);
}

File: app/src/main/kotlin/com/smartsoft/pereverzev/dagger2newtutorial/screens/HomeActivityComponent.kt
package com.smartsoft.pereverzev.dagger2newtutorial.screens

import com.smartsoft.pereverzev.dagger2newtutorial.GithubApplicationComponent
import dagger.Component

/**
 * Created by alexander on 03.01.2018.
 */
@HomeActivityScope
@Component(modules = [HomeActivityModule::class], dependencies = [GithubApplicationComponent::class])
interface HomeActivityComponent {
    fun injectHomeActivity(homeActivity: HomeActivity)
}